Mudraptorbeak.xml

```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<huskappendage>
+  <limb id="13" name="Beak0" type="None" inheritlimbdepth="Head" width="0" height="0" notes="" flip="True" healthindex="0" attackpriority="0" steerforce="0" stepoffset="0,0" radius="50" mass="0" density="0" pullpos="0,0" refjoint="-1" ignorecollisions="False" friction="0.3" restitution="0.05" spriteorientation="90" mirrorvertically="False" mirrorhorizontally="False" hide="False" angulardamping="100" mouthpos="0,0" constanttorque="0" constantangle="0" scale="0.5" attackforcemultiplier="1" minseverancedamage="0.8" severedfadeouttime="0" applytailangle="False" applysinemovement="True">
+    <sprite sourcerect="410,18,97,67" origin="0.29,0.6" depth="0.049" color="180,115,115,255" deadcolor="255,255,255,255" deadcolortime="0" texture="Content/Items/Genetic/Genetic.png" ignoretint="true" />
+    <!-- hides the limb using conditions -->
+    <StatusEffect type="Always" target="This" interval="1" >

```

Comment:
This feels like a somewhat hacky way to go about this. Why does the effect need to run every second? Why does it hide the limb for 100000 seconds instead of indefinitely? I think it would be preferable to only run this once (I'm not sure if there's existing support for that, but could be done e.g. by making OnSpawn effects run on the limbs when the limb spawns) and hide the limb forever (can be done just by omitting the duration).

Reply:
If I recall, it's to refresh the beak when you equip or unequip genes. 

You can only check the genetic affliction to determine whether to hide the limb or not, so the statuseffect does it regularly. I can't think of another way.

As you said the duration of 100000 is basically forever, couldnt think of  another way to do it

Will run some tests

Reply:
I have tried both:

`    <StatusEffect type="OnSpawn" target="This,Limb" >`
which doesn't do anything

`    <StatusEffect type="OnActive" target="Limb" >`
which does work but probably runs every second, and also causes issues with advanced gene splicer if you add two mudraptor genes that take you over the threshold

not specifying duration sends an error message

Reply:
Yeah, OnSpawn effects don't work on limbs atm, I think that should be implemented though.

> not specifying duration sends an error message

That is also something that should be fixed. I'm not seeing any errors when removing the duration though, what kind of an error are you getting?

Reply:
> I still think the advanced gene splicer issue would remain with OnSpawn, if you equip two partial mudraptor genes and the affliction goes over 33/66/100, the limbs become invisible

I don't quite understand why that's an issue with OnSpawn, but not when the effects run constantly. What unhides the "higher-tier" limb when you use Always?

Reply:
I assume the limbs are all spawned at once when you get the mudraptor genes affliction (any value)
It then checks the affliction every second and hides 3 out of 4 limbs based on affliction value

OnSpawn will only check the initial value of the affliction when the gene is equipped.
If the affliction value is changed by adding a 2nd mudraptor gene in Advanced Splicer slot 2, the value changes, but since the limb is hidden forever "OnSpawn" (with no duration) it cannot appear again. 

OnSpawn only works if you completely remove the gene, which is not a problem with normal splicer, but an issue with advanced splicer

Reply:
> If the affliction value is changed by adding a 2nd mudraptor gene in Advanced Splicer slot 2, the value changes, but since the limb is hidden forever "OnSpawn" (with no duration) it cannot appear again.

I understand that, but what unhides the limb when you use Always?

Reply:
The fact that it runs on an interval so it checks the conditionals I assume? 

I honestly don't know but it works that way 😔

Reply:
It hides the limb every x seconds, but it doesn't seem like the limb is ever unhidden.

Reply:
I gave the latest version a test now, and it seems the limbs are never unhidden. If you put a let's say 50% quality mudraptor gene in an advanced gene splicer, and then add another one, no limbs are visible (because the highest-tier limb is hidden and there's nothing that'd unhide it).

I feel like getting this to work robustly might require some code changes. Another option could be to only hide the limbs for some duration (one second for example), and re-check the conditions periodically (with a slightly smaller interval than the limbs are hidden).

Reply:
True, I think it's a bigger issue with advanced gene splicer which doesn't "update" when you insert new genes, I've tried several methods

Hiding limbs is possible, unhiding them isnt

Reply:
> I feel like getting this to work robustly might require some code changes. Another option could be to only hide the limbs for some duration (one second for example), and re-check the conditions periodically (with a slightly smaller interval than the limbs are hidden).

Tried 

    <StatusEffect type="Always" target="Limb" interval="0.5">
      <HideLimb duration="1" />
      <conditional naturalmeleeweapon="gt 33" />
    </StatusEffect>
    
    
    No luck